### PR TITLE
Loading KTX2 texture compatibility: When the extension is not set, th…

### PR DESCRIPTION
**Pull request by bahhchen**
…e default value can be used.

Related issue: #XXXX

**Description**

A clear and concise description of what the problem was and how this pull request solves it.

*This contribution is funded by [Example](https://example.com)*

### CODE REVIEW

**Comment by Mugen87** [comment]
Can you please share a glTF asset that demonstrates the current restriction in `GLTFLoader`. 

I'm not sure I understand your fix...

**Comment by bahhchen** [comment]
> Can you please share a glTF asset that demonstrates the current restriction in `GLTFLoader`?
> 
> I'm not sure I understand your fix...

glTF asset: examples/models/gltf/Building.glb

**Comment by Mugen87** [comment]
It seems you glTF asset is not spec conform. When validating, I see this:

> EXTURE_INVALID_IMAGE_MIME_TYPE	Invalid MIME type 'image/ktx2' for the texture source. Valid MIME types are ('image/jpeg', 'image/png').	/textures/0/source

When a glTF asset holds KTX2 textures, it must use the `KHR_texture_basisu` extension which is missing in your asset definition.

**Comment by bahhchen** [comment]
> It seems you glTF asset is not spec conform. When validating, I see this:
> 
> > EXTURE_INVALID_IMAGE_MIME_TYPE	Invalid MIME type 'image/ktx2' for the texture source. Valid MIME types are ('image/jpeg', 'image/png').	/textures/0/source
> 
> When a glTF asset holds KTX2 textures, it must use the `KHR_texture_basisu` extension which is missing in your asset definition.
> 
> We only support spec conform glTF assets in `GLTFLoader`.

But the official glTF viewer can load it normally: https://github.com/KhronosGroup/glTF-Sample-Viewer

File: examples/jsm/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -1407,13 +1407,19 @@ class GLTFTextureBasisUExtension {
 
 		const textureDef = json.textures[ textureIndex ];
 
+		let extension;
 		if ( ! textureDef.extensions || ! textureDef.extensions[ this.name ] ) {
 
-			return null;
-
+			if (textureDef.source == undefined || json.extensionsRequired != EXTENSIONS.KHR_MATERIALS_UNLIT){
```
Comment (donmccurdy) [inline examples/jsm/loaders/GLTFLoader.js:1413]: Note `json.extensionsRequried` is an array, and will never equal `EXTENSIONS.KHR_MATERIALS_UNLIT`.

**Comment by Mugen87** [comment]
Indeed, glTF sample viewer can load the file.

@donmccurdy What do you suggest in this case?

**Comment by donmccurdy** [comment]
I don't think we should support glTF files using KTX2 textures without the required extension – it would be a difficult to task to guarantee that a viewer *never* loads a file that disobeys the spec (three.js certainly does, but not intentionally...) and trying to expand our scope to support all the invalid cases that happen not to break other engines/viewers will increase complexity.

The KHR_texture_basisu extension should be included on this file — I think this could be reported as a bug to Cesium, as "cesiumlab" appears to be the generator of the file.

File: examples/jsm/loaders/GLTFLoader.js (hunk 1 above)
Comment (bahhchen) [reply examples/jsm/loaders/GLTFLoader.js]: Sorry, the check for extensionsRequired is invalid; I should use array elements instead.

**Comment by bahhchen** [comment]
> I don't think we should support glTF files using KTX2 textures without the required extension – it would be a difficult to task to guarantee that a viewer _never_ loads a file that disobeys the spec (three.js certainly can load some invalid files, we don't validate every input at runtime) and trying to expand our scope to support all the invalid cases that happen not to break other engines/viewers will increase complexity.
> 
> The `KHR_texture_basisu` extension should be included on this file — I think this could be reported as a bug to Cesium, as "cesiumlab" appears to be the generator of the file.

I understand, thank you for the explanation.
Since I'm not very familiar with the GLTF format specifications, I just reviewed the code of glTF-Sample-Renderer and noticed that the texture doesn't use extensions but directly uses its source field.

**Pull request closed without merging**